tools/dds/dds-server/dds-server.cpp

```diff
@@ -115,20 +116,23 @@ void dds_server::post_device_changes(
     }
 }
 
-void dds_server::remove_dds_device( std::string device_name )
+void dds_server::remove_dds_device( const std::string & device_name )
 {
     // deleting a device also notify the clients internally
-    _publisher->delete_datawriter( _devices_writers[device_name].data_writer );
-    std::cout << "Device '" << device_name << "' - removed" << std::endl;
+    auto ret = _publisher->delete_datawriter( _devices_writers[device_name].data_writer );
```

Comment:
I thought the type was `ReturnCode_t` which is a wrapper around `uint32_t`, with `RETCODE_OK` 0.
Doesn't that mean the next `if` should be reversed?
Maybe we should a common function to deal with DDS return codes?